src/data_types/mod.rs

```diff
@@ -117,6 +117,12 @@ pub type PhysicalAddress = u64;
 /// of target platform.
 pub type VirtualAddress = u64;
 
+/// Opaque pointer returned by `BootServices::register_protocol_notify() to be used
+/// with `BootServices::locate_handle` as the `key` parameter.
+#[derive(Debug, Clone, Copy)]
+#[repr(transparent)]
+pub struct SearchKey(NonNull<c_void>);
```

Comment:
A couple suggestions on this type:
1. I think it belongs in the `uefi::table::boot` module since it's both produced and used by methods of `BootServices`.
2. A more specific name would be good, how about `ProtocolSearchKey`?

Reply:
Agreed, both changes made.